src/functions/updateSwapProposal/handler.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type { IValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
+import { formatJSONResponse } from '@libs/api-gateway';
+import { wrapWithConnection } from '@libs/lambda';
+
+import { validatePassword } from '@libs/util';
+import { ApiError, LambdaError } from '@libs/errors';
+import { AUTHPASSWORD_HEADER_KEY } from '@libs/constants';
+import { getProposalFromDb, updateProposalOnDb } from '@services/proposals';
+import { IUpdateProposalRequest } from '@models/update';
+import updateProposalSchema from './schema';
+
+const update: IValidatedEventAPIGatewayProxyEvent<typeof updateProposalSchema> = async (event) => {
+  const {
+    partialTx,
+    signatures,
+    version,
+  } = event.body as IUpdateProposalRequest;
+
+  // Password validation
+  const authPassword = event.headers[AUTHPASSWORD_HEADER_KEY] || '';
+  if (authPassword.length < 3) {
+    throw new LambdaError('Invalid password', ApiError.InvalidPassword);
+  }
```

Comment:
Is the password required or can it be optional?

Reply:
It's required, and must be longer than 3 characters